sqlglot/tokens.py

```diff
@@ -777,6 +779,27 @@ def _scan_number(self):
             else:
                 return self._add(TokenType.NUMBER)
 
+    def _scan_hex_literal(self):
```

Comment:
you may be able to move this whole method into _scan_number

also, can you test out mysql's behavior on how it handles something like 0xbaz ? if it errors out, then we don't need the complicated logic

Reply:
`"Unknown column '0xbaz' in 'field list'`